src/LogicTasks/Syntax/DecomposeFormula.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module LogicTasks.Syntax.DecomposeFormula where
+
+
+import Control.Monad.Output (
+  LangM,
+  OutputMonad, german, english,
+  GenericOutputMonad (paragraph, indent, translatedCode, refuse, image), translate, localise, translations, ($=<<)
+  )
+
+import LogicTasks.Helpers (extra, example, instruct, keyHeading, basicOpKey, arrowsKey, reject)
+import Trees.Types (TreeFormulaAnswer(..))
+import Tasks.DecomposeFormula.Config (DecomposeFormulaInst(..), DecomposeFormulaConfig, checkDecomposeFormulaConfig)
+import Trees.Print (display, transferToPicture)
+import Control.Monad (when, unless)
+import Data.Maybe (isNothing, fromJust)
+import Trees.Helpers (collectLeaves, collectUniqueBinOpsInSynTree, swapKids)
+import Data.Containers.ListUtils (nubOrd)
+import Control.Monad.Cont (MonadIO (liftIO))
+import LogicTasks.Syntax.TreeToFormula (cacheTree)
+
+
+
+
+description :: OutputMonad m => DecomposeFormulaInst -> LangM m
+description DecomposeFormulaInst{..} = do
+
+    example (display tree) $ do
+      english "The following formula is given:"
+      german "Die folgende Formel ist gegeben:"
+
+    instruct $ do
+      english "Create the syntax tree for this formula and swap the two subtrees at the root. "
+      english "Finally, enter the formula for the resulting syntax tree."
+      german "Erstellen Sie den Syntaxbaum zu dieser Formel und tauschen Sie die beiden Teilbäume an der Wurzel. "
+      german "Geben Sie anschließend die Formel für den entstandenen Syntaxbaum ein."
+
+    instruct $ do
+      english "(You are allowed to add arbitrarily many additional pairs of brackets in the formula.)"
+      german "(In der Formel dürfen Sie beliebig viele zusätzliche Klammerpaare hinzufügen.)"
+
+    when addExtraHintsOnAssociativity $ instruct $ do
+      english "Remark: Do not try to use associativity in this task in order to save brackets."
+      german "Hinweis: Sie dürfen bei dieser Aufgabe nicht Klammern durch Verwendung von Assoziativität einsparen."
+
+    keyHeading
+    basicOpKey
+    arrowsKey
+
+    paragraph $ indent $ do
+      translate $ do
+        english "A solution attempt could look like this: "
+        german "Ein Lösungsversuch könnte beispielsweise so aussehen: "
+      translatedCode $ flip localise $ translations $ do
+        english "(A or not B) and C"
+        german "(A oder nicht B) und C"
+      pure ()
+
+    extra addText
+    pure ()
+
+
+verifyInst :: OutputMonad m => DecomposeFormulaInst -> LangM m
+verifyInst _ = pure ()
+
+
+
+verifyConfig :: OutputMonad m => DecomposeFormulaConfig -> LangM m
+verifyConfig = checkDecomposeFormulaConfig
+
+
+
+start :: TreeFormulaAnswer
+start = TreeFormulaAnswer Nothing
+
+
+
+partialGrade :: OutputMonad m => DecomposeFormulaInst -> TreeFormulaAnswer -> LangM m
+partialGrade DecomposeFormulaInst{..} sol = do
```

Comment:
I suggest to rightaway refactor `partialGrade` and `completeGrade` in analogy to what happened for all other task types in e0df94df198077e4f7469dcdd7373eda3f32f138 (the simple cases, not `TreeToFormula`).